onnxruntime/test/framework/inference_session_test.cc

```diff
@@ -946,12 +946,10 @@ TEST(InferenceSessionTests, TestOptionalInputs) {
   // required, optional and invalid input
   status = RunOptionalInputTest(true, true, true);
   ASSERT_FALSE(status.IsOK());
-  EXPECT_THAT(status.ErrorMessage(), testing::HasSubstr("Invalid Feed Input Names: unknown_input"));
 
```

Comment:
How do you know that it failed where you thought it did, if you don't do some kind of test on the error message? Having to update an error message string once in a while seems like a reasonable cost to make sure the test is doing what you think it is.

Reply:
I think the status failure check above is good enough. Maintaining an error message in this way is hard, especially it's also verified again in C# test cases. I'd remove this kind of check.

Reply:
Sorry, I don't understand not checking the error message at all as you're no longer sure it failed for the expected reason. I understand checking partial strings so it it's not as sensitive to change (e.g. just check for "Invalid Feed Input"), but having no check of the error message at all seems too imprecise.

Reply:
Concrete example. When I was making the caching changes the place where the inputs was being checked moved to a different point, and the error message became very unhelpful. I wouldn't have noticed without this check, and it would have resulted in a user passing an invalid input finding it much harder to figure out what they did wrong.

Reply:
ok, will add it back.